madgraph/interface/reweight_interface.py

```diff
@@ -1024,58 +1313,258 @@ def change_kinematics(self, event):
+                    if ij_comb == []:
+                        event_to_sud = buff_event
+                        n_part = nexternal+1
+                        mapped_tag, mapped_order = event_to_sud.get_tag_and_order()
+                        #### H1 type
+                        type = 1
+                    else:
+                        event_for_sud = self.merge_particles_kinematics(buff_event, min_i,min_j,ij_comb)
+                        event_to_sud = event_for_sud
+                        n_part = nexternal 
+                        mapped_tag, mapped_order = event_to_sud.get_tag_and_order()
+                        # map to n+1 body if recoil does not exist at Born level among processes
+                        if mapped_tag not in sud_mod.pdg2ewsud_dict.keys():
+                            event_to_sud = buff_event
+                            n_part = nexternal+1
+                            mapped_tag, mapped_order = event_to_sud.get_tag_and_order()
+                            type = 1   #### H1 type
+                        else:
+                            type = 2    #### H2 type                 
+            elif (len(buff_event) == nexternal): # is an S-event
+                    event_to_sud = buff_event
+                    n_part = nexternal 
+                    mapped_tag, mapped_order = event_to_sud.get_tag_and_order()
+                    type = 0   ### S type
             else:
-                nhel = 0
-            misc.sprint(nhel, Pdir, hel_dict)                        
-            raise Exception("Invalid matrix element for original computation (weight=0)")
+                logger.critical('ERROR: neither H nor S event!')
+                logger.critical(buff_event)
+                sys.exit(2)
+                
+            # Boost to partonic CM frame if not already in one for the momentum reshuffling 
+            E, px, py, pz = 0.,0.,0.,0.
+            for i,particle in enumerate(event_to_sud):
+                    if particle.status == -1:
+                        E += particle.E
+                        px += particle.px
+                        py += particle.py
+                        pz += particle.pz
+            in_part_mom = lhe_parser.FourMomentum([E, px, py, pz])
+            if not ((abs(px) < 1e-6 * E) and (abs(py) < 1e-6 * E) and (abs(pz) < 1e-6 * E)):
+                event_to_sud.boost(in_part_mom)
+
+            # Rotate system to a partonic CM along z-axis
+            initial = copy.deepcopy(event_to_sud[0])
+            if not ((abs(initial.px) < 1e-6 * initial.E) and (abs(initial.py) < 1e-6 * initial.E)):
+                for p in event_to_sud:
+                    p.set_momentum(lhe_parser.FourMomentum(p).rotate_to_z(prot=lhe_parser.FourMomentum(initial)))
+            
+            # Set all light quarks and lepton masses to zero in event file
+            self.set_final_jet_mass_to_zero(event_to_sud)
```

Comment:
```suggestion
            event_to_sud.set_final_jet_mass_to_zero()
```
to be coherent to the move the lhe_parser that I was suggesting

Reply:
should be done